### PR TITLE
Find config and plugin dir

### LINKED ISSUE #158: Attempt to find config file and plugin dir
Instead of asking the user to input their config file and plugin dir in `join`, we should check a few well known locations to see if we can find them and give the user a chance to confirm.

### PR DESCRIPTION
**Pull request by holisticode**
Closes #158

### CODE REVIEW

File: cmd/subnetcmd/join.go [hunk 1]
```diff
@@ -197,9 +242,12 @@ but until the node is whitelisted, it will not be able to validate this subnet.`
 	// or, pluginDir was set but not avagoConfigPath
 	// if **both** flags were set, this will be skipped...
 	if avagoConfigPath == "" {
-		avagoConfigPath, err = app.Prompt.CaptureString("Path to your existing config file (or where it will be generated)")
-		if err != nil {
-			return err
+		avagoConfigPath = findAvagoConfigPath()
+		if avagoConfigPath == "" {
```
Comment (dasconnor) [inline cmd/subnetcmd/join.go:246]: Should we provide an option to override or confirm the found path? Or maybe that would be to just use flags?

**Review by dasconnor — COMMENTED** [review]
Code looks great, just have a question about user confirmation.

File: cmd/subnetcmd/join.go (hunk 1 above)
Comment (holisticode) [reply cmd/subnetcmd/join.go:246]: Hmm. 

I think you're right. We shouldn't just take whatever it found. We should probably give the user a chance to confirm.

Comment (holisticode) [reply cmd/subnetcmd/join.go:261]: Updated with a fix

File: cmd/subnetcmd/join.go [hunk 2]
```diff
@@ -73,6 +74,8 @@ func init() {
 		wd,
 		home,
 		filepath.Join(home, constants.AvalancheGoRepoName),
+		filepath.Join(home, "go", "src", "github.com",
```
Comment (dasconnor) [inline cmd/subnetcmd/join.go:77]: Can we make these constants?

File: cmd/subnetcmd/join.go [hunk 3]
```diff
@@ -243,6 +246,18 @@ but until the node is whitelisted, it will not be able to validate this subnet.`
...
+		if avagoConfigPath != "" {
+			ux.Logger.PrintToUser(logging.Bold.Wrap(logging.Green.Wrap("Found a config file at %s")), avagoConfigPath)
+			yes, err := app.Prompt.CaptureYesNo("Is this the file we should update?")
+			if err != nil {
+				return err
+			}
+			if yes {
+				ux.Logger.PrintToUser("Will use file at path %s to update the configuration", avagoConfigPath)
+			} else {
+				avagoConfigPath = ""
```
Comment (dasconnor) [inline cmd/subnetcmd/join.go:258]: This is a great way to handle this control logic!

**Review by dasconnor — COMMENTED** [review]
One small suggestion, then ready to go

File: cmd/subnetcmd/join.go (hunk 2 above)
Comment (holisticode) [reply cmd/subnetcmd/join.go:77]: We could make constants for each part ("go", "src", etc.) individually. Which seems cumbersome.

But we can't make it a constant due to the filepath.Join() construct. 
I made it a `var` local to this file (and package) for now

**Pull request merged**